File: solutions/python/bob/bob.py
def response(hey_bob):
    hey_bob = hey_bob.strip()
    is_question = hey_bob.endswith('?')
    is_shouting = hey_bob.isupper()

    return "Calm down, I know what I'm doing!" if is_shouting and is_question \
        else "Whoa, chill out!" if is_shouting \
        else "Sure." if is_question \
        else "Fine. Be that way!" if hey_bob == "" \
        else "Whatever."

File: solutions/scala/bob/src/main/scala/Bob.scala
object Bob:
  def response(statement: String): String =
    val trimmed = statement.trim
    val isQuestion = trimmed.endsWith("?")
    val isYelling =
      val letters = trimmed.filter(_.isLetter)
      letters.nonEmpty && letters.forall(_.isUpper)

    trimmed match
      case _ if isYelling && isQuestion =>
        "Calm down, I know what I'm doing!"
      case _ if isYelling =>
        "Whoa, chill out!"
      case _ if isQuestion =>
        "Sure."
      case "" =>
        "Fine. Be that way!"
      case _ =>
        "Whatever."
